documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -194,28 +638,92 @@ The auto config properties for distributed caches are:
-| hibernate.search.default.exclusive_index_use | true | indexing operation in exclusive mode, allowing Hibernate Search to optimize writes
-| hibernate.search.default.indexmanager | org.infinispan.query.indexmanager.InfinispanIndexManager | Delegates index writing to a single node in the Infinispan cluster
-| hibernate.search.default.reader.strategy | shared | Reuse index reader across several queries, avoiding reopening it
+| default.directory_provider | infinispan | Indexes stored in Infinispan. More details at link:http://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#infinispan-directories[Hibernate Search documentation]
+| default.exclusive_index_use | true | indexing operation in exclusive mode, allowing Hibernate Search to optimize writes
+| default.indexmanager | org.infinispan.query.indexmanager.InfinispanIndexManager | Delegates index writing to a single node in the Infinispan cluster
+| default.reader.strategy | shared | Reuse index reader across several queries, avoiding reopening it
 |==============================================================================
 
+===== Re-indexing [[query.massindexer]]
+
+Occasionally you might need to rebuild the Lucene index by reconstructing it from the data stored in the Cache. You need to rebuild the index if you change the definition of what is indexed on your types, or if you change for example some _Analyzer_ parameter, as Analyzers affect how the index is defined. Also, you might need to rebuild the index if you had it destroyed by some system administration mistake. To rebuild the index just get a reference to the MassIndexer and start it; beware if might take some time as it needs to reprocess all data in the grid!
+
+[source,java]
+----
+// Blocking execution
+SearchManager searchManager = Search.getSearchManager(cache);
+searchManager.getMassIndexer().start();
+
+// Non blocking execution
+CompletableFuture<Void> future = searchManager.getMassIndexer().startAsyc();
+
+----
+
+TIP: This is also available as a `start` JMX operation on the link:{javadocroot}/jmxComponents.html#MassIndexer[MassIndexer MBean]
+registered under the name `org.infinispan:type=Query,manager="{name-of-cache-manager}",cache="{name-of-cache}",component=MassIndexer`.
+
+===== Indexless [[query.indexless]]
+
+TODO
+
+===== Hybrid [[query.hybrid]]
 
-==== Lucene Directory
-Infinispan Query isn't aware of where you store the indexes, it just passes the configuration of which _Lucene Directory_ implementation you want to use to the Hibernate Search engine. There are several _Lucene Directory_ implementations bundled, and you can plug your own or add third party implementations: the Directory is the IO API for Lucene to store the indexes.
+TODO
 
-The most common _Lucene Directory_ implementations used with _Infinispan Query_ are:
+===== Mapping Entities
 
+Infinispan relies on Hibernate Search rich API in order to define fine grained configuration for indexing at entity level.
```

Comment:
s/Infinispan relies on Hibernate Search rich API/Infinispan relies on the rich API of Hibernate Search/

And a link to http://hibernate.org/search/ would be appreciated ;-)